Jenkinsfile

```diff
@@ -4,5 +4,16 @@ library("govuk")
 
 node {
   govuk.setEnvar("PUBLISHING_E2E_TESTS_COMMAND", "test-finder-frontend")
-  govuk.buildProject(sassLint: false, publishingE2ETests: true, brakeman: true, rubyLintDiff: false)
+  govuk.buildProject(
+    beforeTest: {
+      stage("Lint Javascript") {
+        sh("npm i standard")
```

Comment:
Have you thought about using a package.json file? that's how it's set up in Content Publisher: https://github.com/alphagov/content-publisher/blob/f1587ab27002c01a45a17b7ded1b627b002b8954/package.json and allows version to be kept consistent. The usage of ./node_modules/.bin also seems like a bit of a smell since it's calling into a hidden directory

Yarn is used too since rails has in built support for that over NPM: https://github.com/alphagov/content-publisher/blob/master/Jenkinsfile#L11-L14

Reply:
I did think about that, but as I only want to use standardjs (no other libs, all vanilla application code) it seems an unnecessary overhead. I really just want to `npx standard` across the repo on CI, but `npx` isn't a thing on our Jenkins setup.

Agreed about the code smell, I cried a little writing that.

On your recommendation I'm now using `yarn`, which looks cleaner 😄 I think long term I want to add a PR to https://github.com/alphagov/govuk-jenkinslib so that applications can hook into standardjs linting with a simple `javascriptLint: true`.

Reply:
>  I think long term I want to add a PR to https://github.com/alphagov/govuk-jenkinslib so that applications can hook into standardjs linting with a simple `javascriptLint: true`.

That is part of a long term plan too. It's raised in a recent RFC: https://github.com/alphagov/govuk-rfcs/blob/master/rfc-100-linting.md (also note the bit about all apps have a package.json).

One thing I think that is missing here is ability to lint locally - I'm not sure if finder frontend has a rake task for that but this is the content publisher one: hhttps://github.com/alphagov/content-publisher/blob/f1587ab27002c01a45a17b7ded1b627b002b8954/lib/tasks/lint.rake#L8

I can imagine npx would have been nicer :) we could always install it onto our boxes but I think that does move us away from a consistent destination.

Reply:
Thanks, great to know the background! So according to the [implementation steps](https://github.com/alphagov/govuk-rfcs/blob/master/rfc-100-linting.md#implementation), looks like we want `package.json` to handle both `standard` JS and `sass-lint`. I sense the risk that this PR could grow out of scope, so here are the options AFAIK:

1. Go ahead with PR as-is, then open another PR at a later date to implement RFC-100 package solution that handles JS and Sass. Make devs run a globally installed `standard` or rely on CI in the meantime.
2. Get a head start on RFC-100 by adding a package and using this to handle the install of `standard` and expose a local linting task to rake, then open another PR at a later date to extend the package.json to handle Sass linting too. This gives devs a command to run local linting without relying on any globals.
3. Implement RFC-100 in this PR, by adding package.json and tasks for JS and Sass linting, and configure the CI scripts such that Sass linting runs the new way and not via [govuk-lint](https://github.com/alphagov/govuk-lint).

What would you suggest?

Reply:
I'd recommend 2 at this point. 

1 varies away from a defined course, whereas 3 requires going a bit above and beyond and I believe there are still prerequisite steps needed for sass linting.

Reply:
Done 👍